### ISSUE TITLE
no-self-assign false positives

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno_lint#281]
```ts
this.level = options.level;
this.loggerName = options.loggerName;
```
Produces `on-self-assign` diagnostic even though it shouldn't
